packages/sdk/src/types/runtime/actor.ts

```diff
@@ -761,18 +752,18 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
 	}
 
 	private _setCollider(collider: Partial<ColliderLike>) {
-		if (this._collider) {
-			unobserve(this._collider);
-			this._collider = undefined;
+		if (!this._collider) {
+			this._collider = new Collider(this, collider);
+			// Actor patching: Observe the collider component for changed values.
+			observe({
+				target: this._collider,
+				targetName: 'collider',
+				notifyChanged: (...path: string[]) => this.actorChanged(...path),
+				// Trigger notifications for every observed leaf node to ensure we get all values in the initial patch.
+				triggerNotificationsNow: true
+			});
 		}
-
-		this._collider = new Collider(this, collider);
-		observe({
-			target: this._collider,
-			targetName: 'collider',
-			notifyChanged: (...path: string[]) => this.actorChanged(...path),
-			// Trigger notifications for every observed leaf node to ensure we get all values in the initial patch.
-			triggerNotificationsNow: true
-		});
+		// Copying the new values will trigger an actor update and enable/update the collider component.
+		this._collider.copy(collider);

```

Comment:
This is actually changing the behavior of what was previously going on here and I actual believe this is incorrect.  We destroyed the old collider so that no leftover values that are not in the new collider partial are reset to default values.  Simply doing a copy here does not replicate that behavior and will produce different colliders if a value is not explicitly set from the incoming collider like, but was set in the previous collider.

Reply:
Enabled and IsTrigger are persisted Unity-side now, so destroying and recreating the UnityCollider keeps the values of those properties. Thus we don't need to patch them again.

---
In reply to: [313523421](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r313523421) [](ancestors = 313523421)

Reply:
We never wanted to persist those nor never did.  Why are we changing this behavior now?

---
In reply to: [313538914](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r313538914) [](ancestors = 313538914,313523421)

Reply:
Summary of discussions: make sure that if setCollider is called, all the fields are either specified or set to default, and make sure that we're creating the minimum number of patches.

---
In reply to: [313559093](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r313559093) [](ancestors = 313559093,313538914,313523421)

Reply:
That was not what was decided on here.  There should be an explicit reset of values in the existing collider that is no longer being replaced to set those to default.  This function accepts a Partial<ColliderLike> and should support that.  We cannot expect a fully filled out collider like to be passed in here, and thus must account for any missing properties in that collider like in this function.  What we had discussed here was:

1. Reset the existing collider to its default values or create a new collider with its default values when there is not a previous collider to reuse
2. Copy in the partial collider like.

---
In reply to: [313648641](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r313648641) [](ancestors = 313648641,313559093,313538914,313523421)

Reply:
You said that your concern was specifically with the use of the `setCollider` APIs. As it's written, all parameters are set or are provided defaults. Isn't that the same thing?

---
In reply to: [315740022](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r315740022) [](ancestors = 315740022,313648641,313559093,313538914,313523421)

Reply:
My understand is we agreed that creating a new Collider was the right approach here, so that previous field values would not be erroneously preserved.

---
In reply to: [315898690](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r315898690) [](ancestors = 315898690,315740022,313648641,313559093,313538914,313523421)

Reply:
With your latest changes, the incoming collider is a Partial again.

---
In reply to: [315939018](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r315939018) [](ancestors = 315939018,315898690,315740022,313648641,313559093,313538914,313523421)

Reply:
It always took a partial. Plus we need it to be so creating colliders in initializers isn't too verbose.

---
In reply to: [316357410](https://github.com/microsoft/mixed-reality-extension-sdk/pull/415#discussion_r316357410) [](ancestors = 316357410,315939018,315898690,315740022,313648641,313559093,313538914,313523421)